pkg/velerorunner/task.go

```diff
@@ -36,68 +55,63 @@ type Task struct {
 //     PlanResources: plan.GetPlanResources(),
 // }
 //
-// completed, err := task.Run()
+// err := task.Run()
+// switch task.Phase {
+//     case Complete:
+//         ...
+// }
 //
 
 // Run the task.
 // Return `true` when run to completion.
-func (t *Task) Run() (bool, error) {
+func (t *Task) Run() error {
 	// Backup
 	err := t.ensureBackup()
 	if err != nil {
-		return false, err
+		return err
 	}
 	if t.Backup.Status.Phase != velero.BackupPhaseCompleted {
-		t.Log.Info(
-			"Waiting for backup to complete.",
-			"owner",
-			t.Owner.GetName(),
-			"backup",
-			t.Backup.Name)
-		return false, nil
+		t.Phase = BackupStarted
+		t.logPhase()
+		return nil
+	} else {
+		t.Phase = BackupCompleted
+		t.logPhase()
```

Comment:
Do you want to return here? Otherwise the phase will get changed at line 83

Reply:
True.  The phase will never be saved in the `Status` but it's more efficient to continue here.